src/components/BookingPage.js

```diff
@@ -0,0 +1,161 @@
+import { useContext, useEffect, useState } from "react";
+import { useLocation } from "react-router-dom";
+import screen from "./screen.svg";
+import whiteSeat from "./whiteSeat.svg";
+import greySeat from "./greySeat.svg";
+import blueSeat from "./blueSeat.svg";
+import {
+  AllSeatsStyle,
+  BookingPageStyle,
+  ConfirmModalStyle,
+  ModalCloseStyle,
+  RowSeatStyle,
+  ScreenStyle,
+  SeatNumberStyle,
+  SeatStyle,
+  SittingSeatStyle,
+} from "../styles/BookingPage.styled";
+import Context from "./context";
+import { GlobalStyle } from "../styles/GLobal.styled";
+import ConfirmBooking from "./ConfirmBooking";
+
+export default function BookingPage(props) {
+  const IMG_URL = "https://image.tmdb.org/t/p/w500";
+
+  const [selectedSeats, setSelectedSeats] = useState([]);
+  let [confirmedSeats, setConfirmedSeats] = useState([]);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [bookingHomeVisible, setBookingHomeVisible] = useState(true);
+  let seatsForPriceCalc = [];
+  const data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+  const seats = [
+    {
+      row: "A",
+      seat: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+    },
+    {
+      row: "B",
+      seat: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+    },
+    {
+      row: "C",
+      seat: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+    },
+    {
+      row: "D",
+      seat: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+    },
+  ];
+  let location = useLocation();
+  // console.log(location.state.movie.id, location.state.movie.poster_path);
+
+  function handleSeatClick(id) {
+    if (selectedSeats.includes(id)) {
+      let newSelectedSeats = selectedSeats.filter((seat) => {
+        return id !== seat;
+      });
+      setSelectedSeats(newSelectedSeats);
+    } else if (selectedSeats.length < 10) {
+      setSelectedSeats((selectedSeats) => [...selectedSeats, id]);
+    } else alert("you can not select more than 10 seats");
+  }
+
+  function confirmBooking() {
+    localStorage.setItem(location.state.movie.id, selectedSeats);
+    setConfirmedSeats(
+      localStorage.getItem(location.state.movie.id).split(",").join(", ")
+    );
+    setModalVisible(true);
+    setBookingHomeVisible(false);
+  }
+
+  function closeModal() {
+    setModalVisible(false);
+    setBookingHomeVisible(true);
+  }
+
+  useEffect(() => {
+    if (localStorage.getItem(location.state.movie.id) !== null) {
+      setSelectedSeats(
+        localStorage.getItem(location.state.movie.id).split(",")
+      );
+      setConfirmedSeats(
+        localStorage.getItem(location.state.movie.id).split(",").join(",")
+      );
+    }
+  }, []);
+
+  console.log(location.state.movie.id);
+  return (
+    <>
+      <BookingPageStyle
+        style={
```

Comment:
Never use inline styling. Pass the props to styled components and apply this logic inside respective styled file.